ocaml/typing/typeopt.ml

```diff
@@ -518,8 +518,16 @@ and value_kind_record env ~loc ~visited ~depth ~num_nodes_visited
                  sort (we can get this info from the label.ld_layout).  For now
                  we rely on the layout check at the top of value_kind to rule
                  out void. *)
-              value_kind env ~loc ~visited ~depth ~num_nodes_visited
-                label.ld_type
+              match rep with
+              | Record_float | Record_ufloat ->
+                (* We're using the `Pfloatval` value kind for unboxed floats.
```

Comment:
I'm unconvinced here. Are we sure that the optimizations in Flambda2 won't go wrong? It's probably fine (based on the fact that it works for boxed float-records), but this makes me want a double-check from a Flambda2 expert.

Reply:
@mshinwell or @Ekdohibs: could we get an expert opinion?

Reply:
This corresponds to what was already done below for `Record_float`, which forgot the value kind and returned `Pfloatval` for the fields, so it seems safe. Still, it is a bit of a hack, and we might want a proper representation for the `value_kind` of float records at some point.